### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by elmar8287**
- Copy the files calculate.js and operate.js into a logic folder
- Import the files in the Calculator component.
- Implement the event from calculate.js and operate.js
- Practice what you have learned until now: 
- use state and props
- pass props to child components
- use lifecycle methods
- lift the state up.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,32 +1,50 @@
...
+          <button type="button" onClick={(ev) => this.handleClick(ev.target.name)}>1</button>
+          <button type="button" onClick={(ev) => this.handleClick(ev.target.name)}>2</button>
+          <button type="button" onClick={(ev) => this.handleClick(ev.target.name)}>3</button>
+          <button type="button" className="orange" onClick={(ev) => this.handleClick(ev.target.name)}>+</button>
+          <button type="button" className="zero" onClick={(ev) => this.handleClick(ev.target.name)}>0</button>
+          <button type="button" onClick={(ev) => this.handleClick(ev.target.name)}>.</button>
+          <button type="button" className="orange" onClick={(ev) => this.handleClick(ev.target.name)}>=</button>
         </section>
       </div>
     );
```
Comment (DeliceLydia) [inline src/components/Calculator.js]: - [x] Please fix your `click event` currently am not able to do anything with your calculator and this project was about working on the events

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+export default function calculate(obj, buttonName) {
+  if (buttonName === 'AC') {
+    return {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
```
Comment (DeliceLydia) [inline src/logic/calculate.js:14]: - [x] Consider displaying a default zero when you open an application and after performing an operation when you click `AC` you can display a zero

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @elmar8287, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (elmar8287) [reply src/components/Calculator.js]: Done

File: src/logic/calculate.js (hunk 2 above)
Comment (elmar8287) [reply src/logic/calculate.js:14]: Dear @DeliceLydia,

I have fixed your notes. Thanks for review 🥇

**Review by vichuge — APPROVED** [review]
# Approved project! ✅ 🥳 🎈 🎉🎉🎉🎉🎉🎉🎉🎉 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

InbornCandidHornet-size_restricted

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by elmar8287** [comment]
Dear @vichuge 🥇 

Thank you for your review.

**Pull request merged**